File: src/Other/MaterialDesign_Styles_Kit/TextToPlaceholderTextVisibilityConverter.vb
Imports System.Windows
Imports System.Windows.Data
Imports System.Globalization

Namespace Global.MaterialDesign_Styles_Kit
    Public Class TextToPlaceholderTextVisibilityConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            If TypeOf value Is String Then
                If String.IsNullOrEmpty(CStr(value)) Then Return Visibility.Visible
                Return Visibility.Collapsed
            End If
            Throw New InvalidOperationException("The IValueConverter expects a value of type String.")
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace

File: src/OpenSilver.Showcase/Other/MaterialDesign_Styles_Kit/TextToPlaceholderTextVisibilityConverter.cs
using System;
using System.Windows;
using System.Windows.Data;
using System.Globalization;

namespace MaterialDesign_Styles_Kit
{
    public class TextToPlaceholderTextVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value is string)
            {
                if (string.IsNullOrEmpty((string)value))
                    return Visibility.Visible;
                return Visibility.Collapsed;
            }
            throw new InvalidOperationException("The IValueConverter expects a value of type String.");
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}
